.github/workflows/pulumi-preview.yml

```diff
@@ -0,0 +1,46 @@
+name: Pulumi Preview
+
+on:
+  pull_request:
+    branches:
+      - main
+
+permissions:
+  contents: read
+  pull-requests: write
+
+jobs:
+  pulumi-preview:
+    name: Pulumi Preview
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.12'
+
+      - name: Install Rust toolchain
```

Comment:
do we need the rust toolchain for this?

Reply:
Ya I believe so because we're using the cargo owner LocalCommand. But I think I should change to using rustup instead of action-rs, since it's no longer maintained. Will fix.